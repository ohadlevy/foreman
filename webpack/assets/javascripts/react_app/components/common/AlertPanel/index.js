import React from 'react';
import getAlertClass from './Alert.consts';
import Icon from '../Icon/';
import Button from '../forms/Button';

export default ({
  className = '',
  type,
  children,
  title,
  onClose
}) => {
  const CloseButton = ({onClick}) => (
    <Button className="close" aria-hidden="true" onClick={onClick}>
      <Icon type="close" />
    </Button>
  );

  return (
    <div
      className={`${getAlertClass(type, onClose)}${className ? ' ' + className : ''}`}
    >
      {onClose && <CloseButton onClick={onClose} />}
      <Icon type={type} />
      {title && <strong>{title}</strong>}
      {title && <br />}
      {children}
    </div>
  );
};
