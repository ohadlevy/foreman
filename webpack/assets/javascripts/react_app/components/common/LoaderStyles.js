export default {
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexGrow: 1
  }
};
