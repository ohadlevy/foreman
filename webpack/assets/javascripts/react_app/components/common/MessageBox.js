// temporary component
// will be replaced by patternfly markup when available
import React from 'react';

const messageBoxStyles = {
  container: {
    flexGrow: 1,
    flexShrink: 1,
    display: 'flex',
    flexDirection: 'column',
    alignContent: 'center',
    justifyContent: 'center'
  },

  content: {
    flexBasis: '30px',
    textAlign: 'center'
  },

  icon: {
    fontSize: '22px'
  },

  message: {
    color: '#363636',
    fontSize: '12px',
    textTransform: 'capitalize'
  }
};

const MessageBox = ({ msg, icontype, style }) =>
(
    <div style={{ ...messageBoxStyles.container, ...style }}>
      <div className={'pficon pficon-' + icontype}
           style={{ ...messageBoxStyles.content, ...messageBoxStyles.icon }}></div>
      <div style={{ ...messageBoxStyles.content, ...messageBoxStyles.message }}>{msg}</div>
    </div>
  );

export default MessageBox;

