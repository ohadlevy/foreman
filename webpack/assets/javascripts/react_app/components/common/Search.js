import React, { Component } from 'react';
import helpers from '../../common/helpers';
import {Form, FormGroup, ControlLabel, FormControl, Button} from 'react-bootstrap';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = { value: this.props.query || ''};
    helpers.bindMethods(this, ['onChange', 'clearClicked', 'handleSubmit' ]);
  }
  onChange(event) {
    event.preventDefault();
    this.setState({value: event.target.value});
  }
  clearButton() {
    let content;

    if (this.state.value.length > 0) {
      content = (
        <Button className="clear" onClick={this.clearClicked}>
          <span className="pficon pficon-close"></span>
        </Button>
      );
    }
    return content;
  }
  clearClicked(event) {
    event.preventDefault();
    this.setState({value: ''});
  }
  handleSubmit(event) {
    event.preventDefault();
    let query = this.state.value.trim();

    if (!query) {
      return;
    }
    // TODO: send request to the server
    this.setState({value: ''});
  }
  render() {
    return (
      <Form role="form" className="search-pf has-button" onSubmit={this.handleSubmit}>
        <FormGroup className="has-clear" controlId="search1">
          <div className="search-pf-input-group">
            <ControlLabel srOnly={true}>Search</ControlLabel>
            <FormControl
              type="search"
              value={this.state.value}
              placeholder="Search"
              onChange={this.onChange}
              />
            {this.clearButton()}
          </div>
        </FormGroup>
        <FormGroup>
          <Button type="submit" bsStyle="default"><span className="fa fa-search"></span></Button>
        </FormGroup>
      </Form>
    );
  }
}

export default Search; // Donâ€™t forget to use export default!
