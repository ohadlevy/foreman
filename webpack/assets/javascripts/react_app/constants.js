export const ACTIONS = {
  RECEIVED_STATISTICS: 'RECEIVED_STATISTICS',
  STATISTICS_REQUEST_ERROR: 'STATISTICS_REQUEST_ERROR',
  RECEIVED_HOSTS_POWER_STATE: 'RECEIVED_HOSTS_POWER_STATE',
  HOSTS_REQUEST_ERROR: 'HOSTS_REQUEST_ERROR',
  CONTROLLER_ADDED: 'CONTROLLER_ADDED',
  CONTROLLER_REMOVED: 'CONTROLLER_REMOVED',
  DISK_ADDED: 'DISK_ADDED',
  DISK_REMOVED: 'DISK_REMOVED'
};

export const STATUS = {
  PENDING: 'PENDING',
  RESOLVED: 'RESOLVED',
  ERROR: 'ERROR'
};

export const VMStorageVMWare = {
  ControllerTypes: {
    VirtualBusLogicController: 'Bus Logic Parallel',
    VirtualLsiLogicController: 'LSI Logic Parallel',
    VirtualLsiLogicSASController: 'LSI Logic SAS',
    ParaVirtualSCSIController: 'VMware Paravirtual'
  },
  MaxControllers: 4,
  defaultConrollerAttributes: {
    type: 'ParaVirtualSCSIController'
  },
  MaxDisksPerController: 15,
  defaultDiskAttributes: {
    size: 0
    // TODO: add more attributes here.
  }
};
