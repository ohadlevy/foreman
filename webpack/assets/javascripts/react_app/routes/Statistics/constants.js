import { foremanUrl } from '../../../foreman_tools';

export const STATISTICS_PAGE_META_RESOLVED = 'STATISTICS_PAGE_META_RESOLVED';
export const STATISTICS_PAGE_META_FAILED = 'STATISTICS_PAGE_FETCH_FAILED';
export const STATISTICS_PAGE_HIDE_LOADING = 'STATISTICS_PAGE_HIDE_LOADING';
export const STATISTICS_DATA_REQUEST = 'STATISTICS_DATA_REQUEST';
export const STATISTICS_DATA_SUCCESS = 'STATISTICS_DATA_SUCCESS';
export const STATISTICS_DATA_FAILURE = 'STATISTICS_DATA_FAILURE';

export const STATISTICS_PAGE_URL = foremanUrl
  ? foremanUrl('/statistics')
  : '/statistics';
