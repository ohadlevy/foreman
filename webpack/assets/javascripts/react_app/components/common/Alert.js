import React from 'react';
import Icon from './Icon';
import { ALERT_CSS } from '../../constants';

/* eslint-disable max-params */
const Alert = ({children, type = 'info', dismissable = false, css = null}) => {
  let classNames = css ? ALERT_CSS[type] + ' ' + css : ALERT_CSS[type];

  if (dismissable) {
    classNames += ' ' + ALERT_CSS.dismissable;
  }

  const closeBtn = (
    <button className="close" data-dismiss="alert" aria-hidden="true">
      <Icon type="close" />
    </button>
  );

  return (
    <div className={classNames}>
      {dismissable && closeBtn}
      {children}
    </div>
  );
};

export default Alert;
