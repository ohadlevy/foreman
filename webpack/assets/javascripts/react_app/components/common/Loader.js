import React from 'react';
import styles from './LoaderStyles';
import {STATUS} from '../../constants';
import MessageBox from './MessageBox';

const Loader = ({ status, children }) => {
let content;

  switch (status) {
    case STATUS.PENDING: {
      content = <div className="spinner spinner-lg"></div>;
      break;
    }
    case STATUS.RESOLVED: {
      content = children[0];
      break;
    }
    case STATUS.ERROR: {
      content = children[1];
      break;
    }
    default:
      content = (<MessageBox icontype="error-circle-o" />);
      break;
  }

  return (
    <div style={styles.root}>
      {content}
    </div>
  );
};

export default Loader;
