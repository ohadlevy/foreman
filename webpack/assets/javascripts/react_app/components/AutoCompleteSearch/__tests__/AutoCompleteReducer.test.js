import reducer from '../AutoCompleteReducer';
import { testReducerSnapshotWithFixtures } from '../../../common/testHelpers';
import {
  AUTO_COMPLETION_REQUEST,
  AUTO_COMPLETION_FAILURE,
  AUTO_COMPLETION_SUCCESS,
} from '../AutoCompleteConstants';
import { STATUS } from '../../../constants';

const getMockOptions = () => [
  {
    completed: '',
    part: ' hardware_model ',
    label: ' hardware_model ',
    category: '',
  },
  {
    completed: '',
    part: ' info ',
    label: ' info ',
    category: '',
  },
  {
    completed: '',
    part: ' name ',
    label: ' name ',
    category: '',
  },
  {
    completed: '',
    part: ' vendor_class ',
    label: ' vendor_class ',
    category: '',
  },
  {
    completed: '',
    part: ' not',
    label: ' not',
    category: 'Operators',
  },
  {
    completed: '',
    part: ' has',
    label: ' has',
    category: 'Operators',
  },
];

const getMockError = () => new Error('Oops');

const fixtures = {
  'should return the initial state': {},
  'should handle AUTO_COMPLETION_REQUEST': {
    action: {
      type: AUTO_COMPLETION_REQUEST,
      payload: {
        options: [],
        error: null,
        status: STATUS.PENDING,
      },
    },
  },
  'should handle AUTO_COMPLETION_SUCCESS': {
    action: {
      type: AUTO_COMPLETION_SUCCESS,
      payload: {
        options: getMockOptions(),
        searchQuery: '',
        error: null,
        status: STATUS.RESOLVED,
      },
    },
  },
  'should handle AUTO_COMPLETION_FAILURE': {
    action: {
      type: AUTO_COMPLETION_FAILURE,
      payload: {
        error: getMockError(),
        status: STATUS.ERROR,
      },
    },
  },
};

describe('AutoCompleteSearch reducer', () => testReducerSnapshotWithFixtures(reducer, fixtures));
