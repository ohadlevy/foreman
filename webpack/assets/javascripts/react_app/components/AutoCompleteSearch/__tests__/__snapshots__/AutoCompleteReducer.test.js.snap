// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteSearch reducer should handle AUTO_COMPLETION_FAILURE 1`] = `
Object {
  "error": [Error: Oops],
  "options": Array [],
  "searchQuery": "",
  "status": "ERROR",
}
`;

exports[`AutoCompleteSearch reducer should handle AUTO_COMPLETION_REQUEST 1`] = `
Object {
  "error": null,
  "options": Array [],
  "searchQuery": "",
  "status": "PENDING",
}
`;

exports[`AutoCompleteSearch reducer should handle AUTO_COMPLETION_SUCCESS 1`] = `
Object {
  "error": null,
  "options": Array [
    Array [
      Object {
        "category": "",
        "completed": "",
        "label": " hardware_model ",
        "part": " hardware_model ",
      },
      Object {
        "category": "",
        "completed": "",
        "label": " info ",
        "part": " info ",
      },
      Object {
        "category": "",
        "completed": "",
        "label": " name ",
        "part": " name ",
      },
      Object {
        "category": "",
        "completed": "",
        "label": " vendor_class ",
        "part": " vendor_class ",
      },
      Object {
        "category": "Operators",
        "completed": "",
        "label": " not",
        "part": " not",
      },
      Object {
        "category": "Operators",
        "completed": "",
        "label": " has",
        "part": " has",
      },
    ],
    "",
    null,
    "RESOLVED",
  ],
  "searchQuery": undefined,
  "status": "RESOLVED",
}
`;

exports[`AutoCompleteSearch reducer should return the initial state 1`] = `
Object {
  "error": null,
  "options": Array [],
  "searchQuery": "",
  "status": "PENDING",
}
`;
